src/python/pants/core/util_rules/system_binaries.py

```diff
@@ -529,5 +529,98 @@ async def find_tar_wrapper(_: TarBinaryRequest, tar_binary: TarBinary) -> TarBin
     return tar_binary
 
 
+# -------------------------------------------------------------------------------------------
+# Binary shims
+# Creates a Digest with a shim for each requested binary in a directory suitable for PATH.
+# -------------------------------------------------------------------------------------------
+
+
+@dataclass(frozen=True)
+class BinaryShimsRequest:
+    requests: tuple[BinaryPathRequest, ...]
+    rationale: str
+    output_directory: str
+
+    @classmethod
+    def for_binaries(
+        cls, *names: str, rationale: str, output_directory: str, search_path: Sequence[str]
+    ) -> BinaryShimsRequest:
+        return cls(
+            requests=tuple(
+                BinaryPathRequest(binary_name=binary_name, search_path=search_path)
+                for binary_name in names
+            ),
+            rationale=rationale,
+            output_directory=output_directory,
+        )
+
+
+@dataclass(frozen=True)
+class BinaryShims:
+    bin_directory: str
+    digest: Digest
+
+
+def create_shim(bash: str, binary: str) -> str:
```

Comment:
Probably mark this private with `_`, given that you don't want dependees to be calling this directly.